tests/instrument/expr_match.ml.reference

```diff
@@ -99,3 +99,174 @@ let f x =
         (match x with
          | Foo  -> (___bisect_mark___expr_match 36; print_endline "foobar")
          | Bar  -> (___bisect_mark___expr_match 37; print_endline "barbar"))))
+let f x =
+  ___bisect_mark___expr_match 43;
+  (match x with
+   | Foo |Bar  as ___bisect_matched_value___ ->
+       ((((match ___bisect_matched_value___ with
+           | Foo  -> (___bisect_mark___expr_match 41; ())
+           | Bar  -> (___bisect_mark___expr_match 42; ())
+           | _ -> ()))[@ocaml.warning "-8-11"]);
+        print_endline "foo"))
```

Comment:
Would it be possible to change the recursion so that the expression is just promoted to the appropriate patterns?
```OCaml
let f x =
  ___bisect_mark___expr_match 43;
  (match x with
   | Foo -> (___bisect_mark___expr_match 41; ())
   | Bar -> (___bisect_mark___expr_match 42; ()));
   print_endline "foo"
```
I think that you had a good reason at the time, which I admit, I don't think I fully understand. I think that this might give some 'power' users pause when they look at the expanded, covered source, and it is not entirely obvious why the code isn't as simple as their original.